src/helpers/json_to_predreq.py

```diff
@@ -1,12 +1,22 @@
-from jaqpotpy.datasets import JaqpotpyDataset
 import pandas as pd
+from src.helpers.recreate_featurizer import recreate_featurizer
+from jaqpotpy.datasets import JaqpotpyDataset
 
 
-def decode(request, model):
+def decode(request):
     df = pd.DataFrame(request.dataset['input'])
     independent_features = request.model['independentFeatures']
-    smiles_cols = [feature['key'] for feature in independent_features if feature['featureType'] == 'SMILES']
+    # smiles_cols = [feature['key'] for feature in independent_features if feature['featureType'] == 'SMILES']
+    smiles_cols = [feature['key'] for feature in independent_features if feature['featureType'] == 'SMILES'] or None
     x_cols = [feature['key'] for feature in independent_features if feature['featureType'] != 'SMILES']
+
+    if request.model['extraConfig']['featurizers']:
+        featurizer_name = request.model['extraConfig']['featurizers'][0]['name']
+        featurizer_config = request.model['extraConfig']['featurizers'][0]['config']
+        featurizer = recreate_featurizer(featurizer_name, featurizer_config)
```

Comment:
What if we have more than 1 featurizers?

Reply:
On point comment. We should extend this functionality to be able to add as many featurizers as desired, if possible

Reply:
@periklis91 is it already supported in jaqpotpy dataset?

Reply:
Even if it's not, a for here for a single featurizer would have the same effect as what you're doing, instead of hardcoding the [0]. Then if in the future we support it from jaqpotpy (if it's not already supported) then we don't have to go here and change the [0] to a for.

Reply:
@vassilismin I cannot recall atm. Give me a few days to finish my current task and then I will switch to 100% jaqpotpy so I will see it then

Reply:
It changed to: 

```
    featurizers = []
    if request.model["extraConfig"]["featurizers"]:
        for i in range(len(request.model["extraConfig"]["featurizers"])):
            featurizer_name = request.model["extraConfig"]["featurizers"][i]["name"]
            featurizer_config = request.model["extraConfig"]["featurizers"][i]["config"]
            featurizer = recreate_featurizer(featurizer_name, featurizer_config)
            featurizers.append(featurizer)
    else:
        featurizers = None

    dataset = JaqpotpyDataset(
        df=df,
        smiles_cols=smiles_cols,
        x_cols=x_cols,
        task=request.model["task"].lower(),
        featurizer=featurizers[0] if isinstance(featurizers, list) else featurizers,
    )
```

When jaqpotpydataset will support multiple featurizers, we need to change 

 ```featurizer=featurizers[0] if isinstance(featurizers, list) else featurizers```
 
 to 
 
```  featurizer=featurizers if isinstance(featurizers, list) else featurizers```

Reply:
if it already works with featurizers, why not make it already like this? 🤔

Reply:
JaqpotpyDataset, as it is right now, should take as featurizer an object of MolecularFeaturizer class. However, ```featurizers``` is a list that its unique object is a MolecularFeaturizer. So with featurizers[0] i parse the MMolecularFeaturizer to JaqpotpyDataset, not a list. In next steps, we should modify JaqpotpyDataset to take a list of MolecularFeaturizer as input, to support multiple featurizers. Till then we need to parse the first object of the list

Reply:
I am on it. I will starting working again on jaqpotpy tomorrow and it will be one of the first things that I will implement

Reply:
@periklis91 it's ok.  Whenever JaqpotpyDataset is ready for multiple featurizers, we need to do this minor change in inference code

Reply:
ok, sorry just to understand! add a `# TODO` so we don't forget it and leave it like this! Resolving the comment